lib/testcontainers.ex

```diff
@@ -175,14 +175,31 @@ defmodule Testcontainers do
     GenServer.call(name, call, @timeout)
   end
 
-  defp create_ryuk_socket(%Container{} = container) do
+  defp create_ryuk_socket(%Container{} = container, reattempt_count \\ 0)
+       when reattempt_count < 3 do
     host_port = Container.mapped_port(container, 8080)
 
-    :gen_tcp.connect(~c"localhost", host_port, [
-      :binary,
-      active: false,
-      packet: :line
-    ])
+    with {:ok, connected} <-
+           :gen_tcp.connect(~c"localhost", host_port, [
+             :binary,
+             active: false,
+             packet: :line,
+             send_timeout: 10000
```

Comment:
👍

Reply:
@jarlah Do you think using a case here would be better instead of with? As we are only trying to connect which will either give an {:ok, _} or {:error, _} tuple?

Reply:
Could be could be. Open for contributions with cleaner code

Reply:
Im waiting with patch release anyway. So if you make a pr we can look at it

Reply:
Created a PR for it!